Introduce TypeBasedParameterResolver adapter (#1984)
junit-team/junit5: documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

For real-world use cases, check out the source code for the `{MockitoExtension}` and the
`{SpringExtension}`.

A generic`{TypeBasedParameterResolver}` is provided when the type of the parameter to inject is the only condition
for your `{ParameterResolver}` to be executed. The `supportsParameters` method is implemented behind the scene

> and supports parametrized types.

Reviewer: :x: ditto

For real-world use cases, check out the source code for the `{MockitoExtension}` and the
`{SpringExtension}`.

> A generic`{TypeBasedParameterResolver}` is provided when the type of the parameter to

Reviewer: ```suggestion
A generic `{TypeBasedParameterResolver}` is provided when the type of the parameter to
```

inject is the only condition for your `{ParameterResolver}` to be executed.

> The `supportsParameters` method is implemented behind the scene and supports parametrized

Reviewer: ```suggestion
The `supportsParameters` method is implemented behind the scenes and supports parameterized
```

types.

> 

Reviewer: ```suggestion
```
